include/zim/archive.h

```diff
@@ -465,10 +465,7 @@ namespace zim
 
       iterator<order>& operator=(iterator<order>& it)
       {
-        m_file.reset();
-        m_entry.reset();
```

Comment:
You need to reset m_entry and m_idx

Reply:
by resetting `m_idx` you mean to set it equal to `it.m_idx` right?

Reply:
Yes.

m_entry is somehow a cache. We must reset it when we change the iterator.
m_file is in which archive the iterator iterate. This may change but it is unlinky.
m_idx is the more important, it is what entry is pointed, we MUST change it